util/exercise/src/lib.rs

```diff
@@ -197,10 +196,11 @@ fn into_literal(item: &Value, use_maplit: bool) -> Result<String> {
         String(s) => format!("\"{}\"", s),
         Number(_) | Bool(_) => format!("{}", item),
         Array(vs) => {
-            let mut items = Vec::with_capacity(vs.len());
-            for im in vs.iter() {
-                items.push(into_literal(im, use_maplit)?);
-            }
+            let items: Vec<string::String> = vs.par_iter()
```

Comment:
It is not impossible that we may see some benefit from parallelization here, but I'd want to see a benchmark to really be convinced.

Reply:
What's the best way to benchmark this function? I gather that `&Value` is from `serde_json`. What data should I input to the function?

Reply:
I don't recommend using native Cargo benchmarks because you end up jumping through a ton of hoops just to get them working right on stable rust, plus you have to have a nightly install available as well to actually run the benches. It's not really worth it. [Criterion](https://crates.io/crates/criterion) makes the process much simpler, and it's pretty well documented.

`into_literal` gets called when transforming a JSON type into a Rust literal. There are three basic tactics you could use to generate `Value`s to bench in this function:

- make up a bunch of JSON by hand
- use a property-based testing library like [quickcheck](https://github.com/BurntSushi/quickcheck) to generate a bunch of cases automatically
- go to the actual data and extract a fair sample of cases

I recommend the third. Since we know that all of our input data comes from the `canonical-data.json` files found in the [problem-specifications repo](https://github.com/exercism/problem-specifications), and since we only care about arrays for this benchmark, we can extract a fair sample of arrays with the following command line:

```sh
$ ls exercises/*/canonical-data.json | 
  xargs jq -c '.. | .cases?[]? | .input[] | arrays' 2>/dev/null | 
  sort -R | 
  head 
```

The output from that command can be transformed pretty trivially into a list of strings, which the benchmark can parse into JSON. Once you've constructed your input vector of JSON `Value`s, all the bencher needs to do is transform each using `into_literal`. Don't forget to [`black_box`](https://docs.rs/criterion/0.2.10/criterion/fn.black_box.html) the output so the compiler doesn't optimize it away!

Reply:
(If you can be tempted to include a non-random selection in your sample: one of the runs when I was developing that command line came up with a really fun array example for a recursive transformer: `[null,[[[null]]],null,null,[[null,null],null],null]`.)